### PR TITLE
Time Diagnostics can be used with Simulated Time

### PR DESCRIPTION
**Pull request by MarcoLm993**
This PR adds the possibility to provide an external rclcpp::Clock instance to the `TopicDiagnostic` and the `TimeStampStatus` objects at construction time.
This way the `TimeStampStatus::tick` method (and the `TopicDiagnostic::tick` one) can compare the input time to different clock sources (e.g. the one by Gazebo), depending on the rclcpp::Clock instance it was provided with.

### CODE REVIEW

File: diagnostic_updater/include/diagnostic_updater/publisher.hpp [hunk 1]
```diff
@@ -45,6 +45,7 @@
 
 #include "rclcpp/publisher.hpp"
 #include "rclcpp/subscription.hpp"
+#include "rclcpp/clock.hpp"
```
Comment (Karsten1987) [inline diagnostic_updater/include/diagnostic_updater/publisher.hpp:48]: please sort alphabetically.

**Review by Karsten1987 — APPROVED** [review]
one nit, otherwise lgtm.

**Comment by Karsten1987** [comment]
I just noticed you've targeted foxy branch. Can you please target ros2? That way we'll have it in the upcoming galactic release and I'll take care of the backporting to foxy.

**Comment by MarcoLm993** [comment]
I will make a new PR targeting to ros2-devel, since this branch cannot be merged there directly.

**Pull request closed without merging**